### PR TITLE
[MM-45052] Add error checking in insights API

### PR DESCRIPTION
**Pull request by cyrilzhang-mm**
#### Summary

Add error checking in the insights API. Return 400 status code if the URL contains an invalid parameter.

This PR should be merged with https://github.com/mattermost/focalboard/pull/3876 & https://github.com/mattermost/mattermost-plugin-playbooks/pull/1496

#### Ticket Link

[Jira ticket](https://mattermost.atlassian.net/browse/MM-45052)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by plant99 — APPROVED** [review]
Neat!

**Review by BenCookie95 — CHANGES_REQUESTED** [review]
Can we add a test that passes and invalid time range in order to test the new error?

**Review by BenCookie95 — APPROVED** [review]
LGTM!

**Comment by BenCookie95** [comment]
@furqanmlk FYI, these 2 PRs are related, https://github.com/mattermost/focalboard/pull/3876 & https://github.com/mattermost/mattermost-plugin-playbooks/pull/1496, to this. We will have to test those with this one.

This PR should not merge until we have worked on a merge strategy with the 2 other teams. We need to ensure that community gets the latest for boards, playbooks and channels at the same time. If this deploys to community too early, without the playbooks and focalboard PRs being deployed, it will break their respective widgets

**Review by furqanmlk — APPROVED** [review]
Working fine

image

**Comment by BenCookie95** [comment]
@plant99 This has changed since you last reviewed it. Added a new method on the server and switched both of Playbooks and Boards code to use it. Once they're both using the new method we delete the old method.

**Review by BenCookie95 — APPROVED** [review]

File: model/insights.go
Comment (plant99) [inline model/insights.go:264]: I know it'd be a big change for me to request. Could we please rename this function to something like `StartOfDayForTimeRangeWithError`? Open to correction, but this might be confusing later on.

**Review by plant99 — APPROVED** [review]
Other things LGTM, just the one suggestion to rename the new function.

File: model/insights.go
Comment (cyrilzhang-mm) [reply model/insights.go:264]: I think the idea is that once all three PRs are merged, we can get rid of the old function, so it won't be confusing. What do you think?

Comment (plant99) [reply model/insights.go:264]: I agree, but 0/5 on the suggestion :) resolving this for now. good work!

**Review by furqanmlk — APPROVED** [review]
Working as expected

**Comment by amyblais** [comment]
@cyrilzhang-mm Is this needed for a specific upcoming self-hosted release version?

**Comment by cyrilzhang-mm** [comment]
@amyblais It isn't needed for any specific release because it's just a bug fix. It can go in 7.5.

**Pull request merged**
